content/en/blog/_posts/2024-02-29-cri-o-seccomp-oci-artifacts.md

```diff
@@ -0,0 +1,477 @@
+or [the prerelease packages](https://github.com/cri-o/packaging?tab=readme-ov-file#usage).
+
+For the demo, I just ran the `crio` binary from my command line using a single
+node Kubernetes cluster via [`local-up-cluster.sh`](https://github.com/cri-o/cri-o?tab=readme-ov-file#running-kubernetes-with-cri-o).
+Now that the cluster is up and running, let's try a pod without the annotation
+running as seccomp `Unconfined`:
+
+```shell
+cat pod.yaml
+```
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+      securityContext:
+        seccompProfile:
+          type: Unconfined
+```
+
+```shell
+kubectl apply -f pod.yaml
+```
+
+The workload is up and running:
+
+```shell
+kubectl get pods
+```
+
+```txt
+NAME   READY   STATUS    RESTARTS   AGE
+pod    1/1     Running   0          15s
+```
+
+And no seccomp profile got applied if I inspect the container using
+[`crictl`](https://sigs.k8s.io/cri-tools):
+
+```shell
+export CONTAINER_ID=$(sudo crictl ps --name container -q)
+sudo crictl inspect $CONTAINER_ID | jq .info.runtimeSpec.linux.seccomp
+```
+
+```txt
+null
+```
+
+Now, let's modify the pod to apply the profile `quay.io/crio/seccomp:v1` to the
+container:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    seccomp-profile.kubernetes.cri-o.io/container: quay.io/crio/seccomp:v1
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+```
+
+I have to delete and recreate the Pod, because only recreation will apply a new seccomp
+profile:
+
+```shell
+kubectl delete pod/pod
+```
+
+```txt
+pod "pod" deleted
+```
+
+```shell
+kubectl apply -f pod.yaml
+```
+
+```txt
+pod/pod created
+```
+
+The CRI-O logs will now indicate that the runtime pulled the artifact:
+
+```txt
+WARN[…] Allowed annotations are specified for workload [seccomp-profile.kubernetes.cri-o.io]
+INFO[…] Found container specific seccomp profile annotation: seccomp-profile.kubernetes.cri-o.io/container=quay.io/crio/seccomp:v1  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Pulling OCI artifact from ref: quay.io/crio/seccomp:v1  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Trying to read profile from: /var/lib/containers/storage/overlay/c15f3eb1efd9abe0ae973e050c374988bba52c45a53f5c3cd80419a865013fa6/merged/seccomp.json  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Retrieved OCI artifact seccomp profile of len: 18853  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+```
+
+And the container is finally using the profile:
+
+```shell
+export CONTAINER_ID=$(sudo crictl ps --name container -q)
+sudo crictl inspect $CONTAINER_ID | jq .info.runtimeSpec.linux.seccomp | head
+```
+
+```json
```

Comment:
Weirdly, `yaml` will work better than `json` (and all JSON documents are valid YAML). Consider switching?

Reply:
Yep, done :+1: